scripts/externalTests/update_external_repos.sh

```diff
@@ -0,0 +1,120 @@
+#
+# solidity is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with solidity.  If not, see <http://www.gnu.org/licenses/>
+#
+# (c) 2022 solidity contributors.
+#------------------------------------------------------------------------------
+
+set -euo pipefail
+
+target_dir="${1:-$(mktemp -d -t update_external_repos_XXXXXX)}"
+
+function clone_repo
+{
+    local upstream_user="$1"
+    local upstream_repo="$2"
+    local fork_name="${3:-$upstream_repo}"
+
+    if [[ ! -d $fork_name ]]; then
+        git clone "git@github.com:solidity-external-tests/${fork_name}.git" --no-checkout
+    else
+        echo "Reusing existing repo: ${fork_name}."
+    fi
+
+    pushd "$fork_name" > /dev/null
+    git remote rm upstream 2> /dev/null || true
+    git remote add upstream "https://github.com/${upstream_user}/${upstream_repo}"
+    popd > /dev/null
+}
+
+function sync_branch
+{
+    local fork_name="$1"
+    local branch="$2"
+
+    echo "${fork_name}: syncing branch ${branch}..."
+    pushd "$fork_name" > /dev/null
+    git fetch upstream "$branch" --quiet
+    git checkout "$branch" --quiet
+    git merge "upstream/${branch}" --ff-only --quiet
+    git push origin "$branch"
+    popd > /dev/null
+}
+
+echo "Entering ${target_dir}"
+cd "$target_dir"
+
+clone_repo brinktrade        brink-core
+clone_repo dapphub           dappsys-monolithic
+clone_repo element-fi        elf-contracts
+clone_repo ensdomains        ens
+clone_repo ensdomains        ens-contracts
+clone_repo euler-xyz         euler-contracts
+clone_repo gnosis            gp-v2-contracts
+clone_repo gnosis            mock-contract
+clone_repo gnosis            safe-contracts
+clone_repo gnosis            util-contracts
+clone_repo JoinColony        colonyNetwork
+clone_repo mycelium-ethereum perpetual-pools-contracts
+clone_repo OpenZeppelin      openzeppelin-contracts
+clone_repo paulrberg         prb-math
+clone_repo pooltogether      v4-core                   pooltogether-v4-core
+clone_repo smartcontractkit  chainlink
+clone_repo sushiswap         trident
+clone_repo Uniswap           v2-core                   uniswap-v2-core
+clone_repo Uniswap           v3-core                   uniswap-v3-core
+clone_repo wighawag          bleeps
+clone_repo yieldprotocol     yield-liquidator-v2
+
+sync_branch brink-core                master
+sync_branch dappsys-monolithic        master
+sync_branch elf-contracts             main
+sync_branch ens                       master
+sync_branch ens-contracts             master
+sync_branch euler-contracts           master
+sync_branch gp-v2-contracts           main
+sync_branch mock-contract             master
+sync_branch safe-contracts            main
+sync_branch util-contracts            main
```

Comment:
I get an error when syncing util-contracts:
`hint: 'main' matched more than one remote tracking branch.
hint: We found 2 remotes with a reference that matched.`

Reply:
Hmm... let me check.

Reply:
Fixed. It was because `git checkout <branch>` fails to create a tracking branch when it's present on more than one remote. I now switched to explicitly specifying branch name and branch to track. I now also force branch creation when it already exists.